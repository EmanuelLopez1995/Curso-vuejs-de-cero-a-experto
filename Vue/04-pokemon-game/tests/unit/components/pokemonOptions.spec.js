import {shallowMount} from '@vue/test-utils';
import PokemonOptions from "@/components/PokemonOptions";
import { pokemons } from "../mocks/pokemons.mock";


describe('PokemonOptions component', ()=>{

    let wrapper; 

    beforeEach(()=>{
        wrapper = shallowMount(PokemonOptions, {
            props: {
                pokemons: pokemons
            }
        })
    })
    test('debe hacer match con el snapshot', () => { 

        expect(wrapper.html()).toMatchSnapshot();
    }) 

    test('debe de mostrar las 4 opciones correctamente', () => { 
        const lis = wrapper.findAll('li');
        expect(lis.length).toBe(4);
        const [li1,li2,li3,li4] = lis;
        expect(li1.text()).toBe('bulbasaur');
        expect(li2.text()).toBe('ivysaur');
        expect(li3.text()).toBe('venusaur');
        expect(li4.text()).toBe('charmander');
    })

    test('debe de emitir "selection" con sus respectivos valores al hacer click', () => { 

        const [li1,li2,li3,li4] = wrapper.findAll('li');
        li1.trigger('click');
        li2.trigger('click');
        li3.trigger('click');
        li4.trigger('click');
        expect(wrapper.emitted('selection').length).toBe(4); // se emiti√≥ 4 veces selection
        expect(wrapper.emitted('selection')[0]).toEqual([ 1 ]); 
        expect(wrapper.emitted('selection')[1]).toEqual([ 2 ]); 
        expect(wrapper.emitted('selection')[2]).toEqual([ 3 ]); 
        expect(wrapper.emitted('selection')[3]).toEqual([ 4 ]); 
    })
})