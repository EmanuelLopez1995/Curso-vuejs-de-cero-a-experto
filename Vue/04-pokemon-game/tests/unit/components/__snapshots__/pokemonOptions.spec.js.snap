// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions component debe hacer match con el snapshot 1`] = `
<div>
  <div class="options-container">
    <ul>
      <li>bulbasaur</li>
      <li>ivysaur</li>
      <li>venusaur</li>
      <li>charmander</li>
    </ul>
  </div>
</div>
`;
